From 90f2f92a9becd50ac2ac23a57ca80ee6bfb8fe5e Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Fri, 17 Mar 2017 14:37:04 +0800
Subject: [PATCH 0869/1345] dts: a3700: add regulator DT node based on AVS

commit  0eca31d09ed18891692cc00b96ce9c7a993e595f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch adds the AVS regulator DT node, which is supposed to
be used by CPU DVS to change the CPU voltage dynamically.

Change-Id: I0c114f918dccad5a85ad6f63a767d2ffe912f11f
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37534
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Evan Wang <xswang@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-3720-community.dts     |    4 ++++
 arch/arm64/boot/dts/marvell/armada-3720-db.dts     |    4 ++++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi       |    7 +++++++
 3 files changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
index 2b6abb2..c9f5d03 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-community.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -72,6 +72,10 @@
 				status = "okay";
 			};
 
+			reg_avs: avs@11500 {
+				status = "okay";
+			};
+
 			eth0: ethernet@30000 {
 				phy-mode = "rgmii-id";
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 0c2823fe..cec6d1d 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -111,6 +111,10 @@
 			bm_bppi: bm-bppi {
 				status = "okay";
 			};
+			reg_avs: avs@11500 {
+				status = "okay";
+			};
+
 		};
 
 		reg_sata: pwr-sata {
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 02f3506..8bc2b99 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -142,6 +142,13 @@
 				status = "disabled";
 			};
 
+			reg_avs: avs@11500 {
+				compatible = "marvell,armada-3700-avs";
+				reg = <0x11500 0x40>;
+				clocks = <&core_clk 4>;
+				status = "disabled";
+			};
+
 			uart0: serial@12000 {
 				compatible = "marvell,armada-3700-uart";
 				reg = <0x12000 0x200>;
-- 
1.7.9.5

